sympy/functions/special/gamma_functions.py

```diff
@@ -799,11 +799,26 @@ def _eval_as_leading_term(self, x, logx=None, cdir=0):
 
     def fdiff(self, argindex=2):
         if argindex == 2:
-            n, z = self.args[:2]
+            n, z = self.args
             return polygamma(n + 1, z)
         else:
             raise ArgumentIndexError(self, argindex)
 
+    def _eval_nseries(self, x, n, logx, cdir=0):
+        from sympy.series.order import Order
+        s, z = self.args
+        d = z.subs(x, 0) # expansion point
+        e = (z-d).as_leading_term(x).as_coeff_exponent(x)[1]
```

Comment:
`d` should be checked first. It may be infinite or None.

Reply:
Should be resolved now.